content/rust/concepts/strings/strings.md

```diff
@@ -0,0 +1,87 @@
+---
+Title: 'strings'
+Description: 'A collection of characters used to represent text.'
+Subjects:
+  - 'Computer Science'
+  - 'Game Development'
+Tags:
+  - 'Strings'
+CatalogContent:
+  - 'rust-for-programmers'
+  - 'paths/computer-science'
+---
+
+In Rust, **strings** are a fundamental data type, consisting of characters such as letters, spaces, numbers, or symbols. There are two main string types:
+
+|            | `String`                       | `&str`                                 |
+| ---------- | ------------------------------ | -------------------------------------- |
+| Mutability | Mutable (can be changed)       | Immutable (cannot be changed)          |
+| Ownership  | Owned (exists on the heap)     | Borrowed (reference to a `String`)     |
+| Use Case   | When modifications are needed  | For fixed or temporary strings         |
+| Slicing    | Slicing returns `&str` slices. | Slicing returns smaller `&str` slices. |
+
+## Creating `String` and `&str` String Slices
+
+### `String` Type
+
+A String can be created using:
+
+- `String::new()` for an empty, mutable string.
+- `String::from("literal")` for initializing with a known value.
+- `"literal".to_string()` for converting a string slice to a `String`.
+
+### `&str` Type
+
+A `&str` string can be created by:
+
+- Assigning a literal directly: `let message: &str = "Hello, world!";`
+- Slicing a String: `let slice = &message_string[0..5];` for a specific byte range.
+- Borrowing from a String: `let slice: &str = &message_string;` for the entire string.
```

Comment:
Explain the method for creating the `&str` string only, and add the rest of the information like slicing and borrowing in their respective part of the content. Add an example in Rust to demonstrate how to create a `&str` string.